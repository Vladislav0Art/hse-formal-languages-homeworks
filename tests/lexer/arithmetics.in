$expr -> $expr '+' $term | $expr '-' $term | $term
$term -> $term '*' $mult | $term '/' $mult | $mult
$mult -> '(' $expr ')'   | [any number]