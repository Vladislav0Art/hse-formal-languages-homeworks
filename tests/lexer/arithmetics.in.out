LexToken(START,'expr',1,0)
LexToken(NON_TERMINAL,'expr',2,12)
LexToken(RULE,'->',2,18)
LexToken(NON_TERMINAL,'expr',2,21)
LexToken(TERMINAL,'+',2,27)
LexToken(NON_TERMINAL,'term',2,31)
LexToken(OR,'|',2,37)
LexToken(NON_TERMINAL,'expr',2,39)
LexToken(TERMINAL,'-',2,45)
LexToken(NON_TERMINAL,'term',2,49)
LexToken(OR,'|',2,55)
LexToken(NON_TERMINAL,'term',2,57)
LexToken(SEPARATOR,';',2,62)
LexToken(NON_TERMINAL,'term',3,64)
LexToken(RULE,'->',3,70)
LexToken(NON_TERMINAL,'term',3,73)
LexToken(TERMINAL,'*',3,79)
LexToken(NON_TERMINAL,'mult',3,83)
LexToken(OR,'|',3,89)
LexToken(NON_TERMINAL,'term',3,91)
LexToken(TERMINAL,'/',3,97)
LexToken(NON_TERMINAL,'mult',3,101)
LexToken(OR,'|',3,107)
LexToken(NON_TERMINAL,'mult',3,109)
LexToken(SEPARATOR,';',3,114)
LexToken(NON_TERMINAL,'mult',4,116)
LexToken(RULE,'->',4,122)
LexToken(TERMINAL,'(',4,125)
LexToken(NON_TERMINAL,'expr',4,129)
LexToken(TERMINAL,')',4,135)
LexToken(OR,'|',4,141)
LexToken(TERMINAL,'0',4,143)
LexToken(OR,'|',4,147)
LexToken(TERMINAL,'1',4,149)
LexToken(OR,'|',4,153)
LexToken(TERMINAL,'2',4,155)
LexToken(OR,'|',4,159)
LexToken(TERMINAL,'3',4,161)
LexToken(SEPARATOR,';',4,164)
